python/sglang/srt/layers/attention/flashinfer_backend.py

```diff
@@ -649,25 +789,65 @@ def call_begin_forward(
         kv_indptr: torch.Tensor,
         qo_indptr: torch.Tensor,
         use_ragged: bool,
+        forward_batch: ForwardBatch = None,
+        encoder_wrappers: List = None,
     ):
         bs = len(req_pool_indices)
-        kv_indptr[1 : bs + 1] = torch.cumsum(paged_kernel_lens, dim=0)
-        kv_indptr = kv_indptr[: bs + 1]
-        kv_indices = torch.empty(
-            paged_kernel_lens_sum, dtype=torch.int32, device="cuda"
-        )
-        create_flashinfer_kv_indices_triton[(bs,)](
-            self.req_to_token,
-            req_pool_indices,
-            paged_kernel_lens,
-            kv_indptr,
-            kv_start_idx,
-            kv_indices,
-            self.req_to_token.shape[1],
-        )
+        if forward_batch is not None and forward_batch.forward_mode.is_draft_extend():
+            # spec extend update generate arg
+            kv_indices, kv_indptr, kv_last_page_len, qo_indptr = (
+                forward_batch.spec_info.generate_attn_arg_spec_extend(
```

Comment:
I think we can keep the current code, but only pass `forward_batch.ForwardMode` and `spec_info`